packages/wrangler/src/dev/create-worker-preview.ts

```diff
@@ -206,15 +214,19 @@ export async function createPreviewSession(
 			token: string;
 			prewarm: string;
 		};
-		const inspector = new URL(inspector_websocket);
-		inspector.searchParams.append("cf_workers_preview_token", token);
-
+		let inspector: URL | undefined;
+		if (!tailLogs) {
+			inspector = new URL(inspector_websocket);
+			inspector.searchParams.append("cf_workers_preview_token", token);
+		}
 		return {
 			id: crypto.randomUUID(),
 			value: token,
-			host: ctx.host ?? inspector.host,
-			inspectorUrl: switchHost(inspector.href, ctx.host, !!ctx.zone),
+			host: ctx.host ?? inspector?.host ?? switchedExchangeUrl.host,
 			prewarmUrl: switchHost(prewarm, ctx.host, !!ctx.zone),
+			...(!tailLogs && inspector
+				? { inspectorUrl: switchHost(inspector.href, ctx.host, !!ctx.zone) }
+				: {}),
```

Comment:
NIT: this switchHost could have been done in the `if` block where the URL is created. Then if the variable had been called `inspectorUrl` I think this would boil down to:


```suggestion
			inspectorUrl,
```